### ISSUE TITLE
Warning on infinite acos gradients?

### ISSUE THREAD

**Issue by stefdoerr** [pytorch/pytorch#4605]
Just a suggestion. Should there maybe be some kind of warning thrown in the case we get `acos` gradients of `-inf`? I understand that the behavior is correct but as the `inf` propagates through the graph you end up with a `NaN` that can take a decent bit of debugging to detect in a larger program.

```python
import torch
angle = torch.autograd.Variable(torch.FloatTensor([0]), requires_grad=True)
cos = torch.cos(angle)
acos = torch.acos(cos)
print(torch.autograd.grad(acos, cos, retain_graph=True)[0])
print(torch.autograd.grad(acos, angle)[0])
```

**Comment by apaszke** [comment]
A lot of our ops work this way, and adding these checks would cause massive slowdowns. Thanks for the suggestion, but we can't afford that.

**Comment by stefdoerr** [comment]
Ok, thanks anyway :)
